### PR TITLE
Implementation of synthetic table data

### PR DESCRIPTION
**Pull request by snova-codym**
This update will:

- Allow for the generation of synthetic table images from modified tsv formatted tables.
- Create qa pairs for table qa.
- Create ocr prompts and completions.

More functionality will likely need to be extended, such as more control over image augmentations.

### CODE REVIEW

File: yoda/llava_data_prep/src/table_utils.py [hunk 1]
```diff
@@ -0,0 +1,1063 @@
...
+
+        logging.info(f"Saving image to: {img_filepath}")
+        image.save(img_filepath, 'JPEG')
+        
+        # Clean up intermediate files.
+        self._crop_synth_table(str(img_filepath))
+        subprocess.run(['rm', tex_filepath])
+        subprocess.run(['rm', pdf_filepath])
+
+class QAItem(BaseModel):
```
Comment (snova-codym) [inline yoda/llava_data_prep/src/table_utils.py:756]: @snova-francescar

File: utils/sec_utilities/edgar_ingestion.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import yaml
```
Comment (snova-kwasia) [inline utils/sec_utilities/edgar_ingestion.py:1]: Can we please move this to a utils within the yoda  kit repo? This isn't a multi kit util.

File: yoda/llava_data_prep/config.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+llm: 
+    "api": "sambastudio"
+    "do_sample": True
+    "temperature": 0.7
+    "streaming": False
+    "max_tokens_to_generate": 1200
+    "coe": True #set as true if using Sambastudio CoE endpoint
+    "select_expert": "Meta-Llama-3.1-70B-Instruct" #set if using sncloud, SambaStudio CoE llm expert
```
Comment (snova-kwasia) [inline yoda/llava_data_prep/config.yaml:8]: Let's please remove the if using sncloud here, as this is a SambaStudio only kit as far as I'm aware.

File: yoda/llava_data_prep/scripts/create_synthetic_tables.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+parser = argparse.ArgumentParser(description="Test and log synthetic table creation.")
+
+parser.add_argument("--name", type=str, help="Name of run", default="llava_prep")
+parser.add_argument("--num-its", type=int, help="Number of iterations.", default=25)
+parser.add_argument("--split", type=str, help="Split", default="train", choices=["train", "val"])
+
+def main() -> None:
+
+    print(LOGS_PATH)
```
Comment (snova-kwasia) [inline yoda/llava_data_prep/scripts/create_synthetic_tables.py:35]: Please remove print here or turn to logging.info if it's needed.

File: yoda/llava_data_prep/src/generate_tables.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+sys.path.append(repo_dir)
+
+from yoda.llava_data_prep.src.table_utils import TableTools # type: ignore
+DATA_DIRECTORY = os.path.join(kit_dir,"synthetic_data")
+
+table_tools = TableTools()
+
+synth_tables: dict = {}
+
+columns: str = "\\begin{tabular}{| l c c r c c r |}\n\\hline\n"
```
Comment (snova-kwasia) [inline yoda/llava_data_prep/src/generate_tables.py:21]: Should these 2 literal strings perhaps be in a config somewhere? I assume they change with different types of table to be generated?

File: yoda/llava_data_prep/table_templates/sec_edgar/AAPL.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
+synth_tables = []
```
Comment (snova-kwasia) [inline yoda/llava_data_prep/table_templates/sec_edgar/AAPL.py:1]: Similar comment to above. Could these live in a json file for the given company in this case Apple? That way they can be pulled in when necessary?

File: yoda/llava_data_prep/table_templates/sec_edgar/AAPL.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+# Template 1: AAPL_Revenue_simple_columns
+ex1 = {"columns": "\\begin{tabular}{| l c c |}\n\\hline\n",
```
Comment (snova-kwasia) [inline yoda/llava_data_prep/table_templates/sec_edgar/AAPL.py:4]: Could you please make these variable names more descriptive? Not sure what ext1 etc means.

**Review by snova-kwasia — CHANGES_REQUESTED** [review]
Have given a first pass at this and added a few comments, they aren't major changes. Additionally can you add a README.md? This is required for any new kit.

**Comment by snova-luiss** [comment]
Minor suggestions:
- Make sure to add output files to the .gitignore (images, json, sec_data, pt, aux, etc)
- Add system dependencies to the readme

**Comment by snova-codym** [comment]
@snova-luiss Thanks - I will do both of these things tomorrow AM.

File: yoda/llava_data_prep/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,120 @@
+aiohappyeyeballs==2.4.0
```
Comment (snova-kwasia) [inline yoda/llava_data_prep/requirements.txt:1]: Is this a pip freeze or the actual requirements (longer than our base-requirements.txt!)? If possible let's just add the requirements for the kit.

File: yoda/llava_data_prep/README.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+- Create virtual environment.
+- pip install yoda/llava_data_prep/requirements.txt
+- Note any system dependancies or issues installing the requirements.
+- Run yoda/llava_data_prep/notebooks/edgar_ingestion.ipynb and ensure it completes.  Check some of the outputs.
+- Run yoda/llava_data_prep/scripts/generate_tables.py and ensure yoda/synthetic_data/tmp/images/test_table.jpg is created.
+- Run:
+```bash
+python yoda/llava_data_prep/scripts/create_synthetic_tables.py --name test --num-its 4
+```
+and ensure yoda/llava_data_prep/test is created with a subfolder of synthetic images and yoda/llava_data_prep/test/annotations_train.json exists.
```
Comment (snova-kwasia) [inline yoda/llava_data_prep/README.md:115]: Might want to change the language here to - "After completion yoda/llava_data_prep/test is created ....etc"
Currently I read this as something the user needs to ensure vs the output of the previous command.

**Review by snova-kwasia — CHANGES_REQUESTED** [review]
Minor changes.

**Comment by snova-luiss** [comment]
The images and the JSON file (annotations_train.json) should also be added to the .gitignore. Other than that, looks good to me!

**Review by snova-luiss — APPROVED** [review]
lgtm!

**Review by snova-kwasia — APPROVED** [review]
LGTM @snova-codym

**Pull request merged**